pac4j-saml/src/main/java/org/pac4j/saml/client/SAML2Client.java

```diff
@@ -150,24 +150,24 @@ protected void initSAMLContextProvider(final MetadataResolver metadataManager) {
     }
 
     protected MetadataResolver initServiceProviderMetadataResolver(final WebContext context) {
-        this.spMetadataResolver = new SAML2ServiceProviderMetadataResolver(this.configuration.getServiceProviderMetadataPath(),
-                computeFinalCallbackUrl(context),
-                this.configuration.getServiceProviderEntityId(),
+		String finalCallbackUrl = computeFinalCallbackUrl(context);
+		this.spMetadataResolver = new SAML2ServiceProviderMetadataResolver(this.configuration.getServiceProviderMetadataPath(),
+				this.configuration.getServiceProviderMetadataResource(), finalCallbackUrl,
+				(this.configuration.getServiceProviderEntityId().equals("CALLBACK_URL") ? finalCallbackUrl
```

Comment:
In your https://github.com/pac4j/spring-security-pac4j-demo/blob/master/src/main/resources/securityContext.xml you had 

> <property name="serviceProviderEntityId" value="http://localhost:8080/callback?client_name=SAML2Client" />

I basically thought instead I could just refer to CALLBACK_URL and in those cases it would just put the callback url rather than hardcoding the config defined somewhere else.